hyperspy/misc/array_tools.py

```diff
@@ -392,6 +391,17 @@ def numba_histogram(data, bins, ranges):
     hist : array
         The values of the histogram.
     """
+    if data.dtype.byteorder == '>':
```

Comment:
What about the following instead?

```python
    if data.dtype.byteorder == '>':
       data = data.byteswap().newbyteorder()
    return _numba_histogram(data, bins, ranges)
```
It is the same, but easier to maintain...

Reply:
Sure, no problem. 

Should I add it for the other 4 numba-functions? I think we cannot rule out that someone calls them with the wrong endian. For the private functions I could just add it to the calling functions, only `lowess` would need a wrapper.

Reply:
The issue is that numba only support native endianess and this approach will not work on all platform. What's about:
```python
        # Numba only supported native dtype
        # https://github.com/numba/numba/issues/2243
        if not data.dtype.isnative:
            data = data.astype(data.dtype.type
```

Reply:
Yes, that should do the job, indeed.